src/lib/optimizer/optimizer.cpp

```diff
@@ -77,7 +78,7 @@ void collect_select_expressions_by_lqp(SelectExpressionsByLQP& select_expression
 namespace opossum {
 
 std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
-  auto optimizer = std::make_shared<Optimizer>(10);
+  auto optimizer = std::make_shared<Optimizer>(100);
```

Comment:
is that really needed, that seems very high. I guess this is mostly due to the fact that the push down rule only pushes down one predicate on each iteration instead of pushing down all predicates that are pushdownable?

If so, then the ppr should be adjusted at some point.

Reply:
Even worse - it only pushes one predicate one step. If there are two projections, that's two iterations.

I already have that on the seminar's topic list, but as long as the optimizer is under "also-ran", I am not too concerned about it.

Reply:
(And that was already part of the old rule. Changing it to make bigger jumps would have also meant changing the push-down-to-join part.)

Reply:
:(